build-in-container/scripts/build-mariner.sh

```diff
@@ -18,6 +18,51 @@ while (( "$#" )); do
             exit 1
         fi
         ;;
+        --RPM_repo_file)
```

Comment:
This argument parsing is pretty unwieldy, but honestly, a lot of bash argument parsing seems difficult to really read through in detail. Consider if this argument parsing could be done in a more readable way in a future PR refactor.

Reply:
@christopherco do we have an example parsing we prefer in Mariner? I am using the template we have in generate_source_tarball.sh

Reply:
I am not aware of one preferred example, which I think contributes to this problem. Tagging some folks to opine on best practices for parsing arguments in bash - @dmcilvaney @tobiasb-ms @corvus-callidus @cwize1 @PawelWMS @nicogbg @anphel31